File: tao/python/118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans =  [[1]]     
        if numRows == 1:
            return ans
    
        for i in range(2, numRows+1):
            curr = [0] + ans[-1] + [0]
            row = []
            for j in range(1, len(curr)):
                row.append(curr[j-1] + curr[j])

            ans.append(row)
           
        return ans 

File: hao/c++/118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> result;
        vector<int> tmp;
        tmp.push_back(1);
        result.push_back(tmp);
        
        if (numRows == 1) {
            return result;
        }
        tmp.push_back(1);
        result.push_back(tmp);
        for (int i=3; i <= numRows; i++) {
            vector<int> temp;
            temp.push_back(1);
            for (std::vector<int>::iterator it = tmp.begin() ; it != tmp.end()-1; ++it) {
                int res = *it + *(it+1);
                temp.push_back(res);
            }
            temp.push_back(1);
            result.push_back(temp);
            tmp = temp;
        }
        return result;
    }
};